src/data/error.rs

```diff
@@ -79,3 +80,136 @@ impl<T, E> Recoverable for Result<T, (E, T)> {
         }
     }
 }
+
+pub enum CompileErrorLevel {
+    Warning,
+    RecoverableError,
+    FatalError,
+}
+
+pub trait NewCompileError {
+    // Idealy this should be an associated constant, but those are not allowed
+    // in trait objects
+    fn level(&self) -> CompileErrorLevel;
+
+    // Could also be an associated const
+    fn name(&self) -> &'static str;
+
+    fn message(&self) -> String;
+    fn location(&self) -> Location;
+}
+
+impl fmt::Debug for (dyn NewCompileError + 'static) {
```

Comment:
Why is this `'static`? That seems very limiting in case we want to reference a type or something.

Reply:
I think I did this because one of the errors said `Debug` (and others) were not implemented for `(dyn NewCompileError + 'static)` so I just implemented it with static without thinking about it.